### PR TITLE
test: MM-T963 Test for Purge Caches

### PR DESCRIPTION
**Pull request by alejandrovelez7**
#### Summary

E2E test around the Purge Caches functionality

#### Ticket Link

Addressing https://github.com/mattermost/mattermost-server/issues/18185

#### Release Note
```
NONE
```

### CODE REVIEW

File: e2e/cypress/integration/system_console/environment_spec.js [hunk 1]
```diff
@@ -284,6 +284,16 @@ describe('Environment', () => {
...
     });
 
+    describe('System Console/Environment/Web Server', () => {
+        it('MM-T963 Configuration - Purge caches', () => {
+            cy.visit('/admin_console/environment/web_server');
+
+            cy.get('#PurgeButton').scrollIntoView().should('be.visible').within(() => {
+                cy.findByText('Purge All Caches').should('be.visible').click().wait(TIMEOUTS.ONE_SEC);
+            });
+        });
```
Comment (alejandrovelez7) [inline e2e/cypress/integration/system_console/environment_spec.js:294]: PR ask was to confirm `Nothing crashes` lmk if there's already a standard way to assert on that! 😄

Comment (josephbaylon) [reply e2e/cypress/integration/system_console/environment_spec.js:294]: @alejandrovelez7 Thanks for working on this e2e. There isn't really a standard way of asserting no crash was found, but you can probably just include additional step to reload the page and assert that the page loaded fine, like asserting on the admin consoler header `Web Server`

Comment (alejandrovelez7) [reply e2e/cypress/integration/system_console/environment_spec.js]: Applied that suggestion in https://github.com/mattermost/mattermost-webapp/pull/9023/commits/93b3dda6a5e27c25cb281f07ab93d28c49c76a42 and spec is passing

image

Let me know if you have any additional feedback!

File: e2e/cypress/integration/system_console/environment_spec.js [hunk 2]
```diff
@@ -284,6 +284,20 @@ describe('Environment', () => {
         });
     });
 
+    it('MM-T963 Configuration - Purge caches', () => {
+        cy.visit('/admin_console/environment/web_server');
+
```
Comment (furqanmlk) [inline e2e/cypress/integration/system_console/environment_spec.js:289]: Thanks @alejandrovelez7  all look good, test passed for me.
Just few small changes to follow standard. 
As mentioned at the top of spec file, can you please add comments using following style,
`#` indicates a test step (e.g. # Go to a page)
`*` indicates an assertion (e.g. * Check the title)

For reference, see MM-T959 in the same spec file.

**Review by furqanmlk — CHANGES_REQUESTED** [review]

File: e2e/cypress/integration/system_console/environment_spec.js (hunk 2 above)
Comment (alejandrovelez7) [reply e2e/cypress/integration/system_console/environment_spec.js:289]: Updated in https://github.com/mattermost/mattermost-webapp/pull/9023/commits/a7ed7af529f8ff553162d41776b588051ad3fd17 let me know what you think!

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @alejandrovelez7 !!

**Review by furqanmlk — APPROVED** [review]
Thanks @alejandrovelez7 :)

Looks good

**Comment by emilyacook** [comment]
Thanks for participating in Hacktoberfest! You can claim your sticker set here: https://get.printfection.com/hacktober21/4144583267 @alejandrovelez7

**Pull request merged**